assets/js/setup.js

```diff
@@ -401,6 +423,36 @@ function flipBoard() {
     adjustSquareSize(scaleOption); // ...and also mess up with piece image sizes, so we resize them
 }
 
+function clockCountdown(first) {
+    // TODO: make this agnostic from PGN tag Reference time
+    let sideToMove = (PlyNumber % 2) === 0;
+    let lastRefreshed = undefined;
+    if (referenceTime) {
+        lastRefreshed = new Date(referenceTime.slice(2));
+    } else {
+        lastRefreshed = new Date(LiveBroadcastLastReceivedLocal);
+    }
+    // let lastRefreshed = LiveBroadcastLastModified;
+    let now = new Date();
+    let timeElapsed = now - lastRefreshed;
+    let activeClk = document.getElementById(sideToMove ? "GameWhiteClock" : "GameBlackClock");
+    let inactiveClk = document.getElementById(!sideToMove ? "GameWhiteClock" : "GameBlackClock");
+    let clockStringRegExp = new RegExp("(\\d+):(\\d+):(\\d+)");
+    if (clkMatch = activeClk.innerHTML.match(clockStringRegExp)) {
+        let h = parseInt(clkMatch[1]),
+            m = parseInt(clkMatch[2]),
+            s = parseInt(clkMatch[3]);
+        let clockTime = new Date((new Date()).setHours(h, m, s) - (first ? timeElapsed : 1000));
+        let clockZero = new Date(new Date().setHours(0, 0, 0));
+        if (clockTime < clockZero) {
+            clockTime = clockZero;
+        }
```

Comment:
For this change we need to wrap it all once again in Date constructor. I don't think it will improve performance nor readability.
`clockTime = new Date(Math.max(clockTime, clockZero));`